### PR TITLE
feat: add bsky.app support

### PR DESCRIPTION
**Pull request by Misaka13514**

### CODE REVIEW

File: nazurin/sites/bluesky/api.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+class Bluesky:
+    @network_retry
+    async def resolve_handle(self, handle: str):
+        """
+        Get the DID from handle.
+        https://www.docs.bsky.app/docs/api/com-atproto-identity-resolve-handle
+        """
+        api = "https://public.api.bsky.app/xrpc/com.atproto.identity.resolveHandle"
+        async with Request(headers={"Accept": "application/json"}) as request:
```
Comment (y-young) [inline nazurin/sites/bluesky/api.py:21]: Why is this header needed?

File: nazurin/sites/bluesky/api.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        """Get all images in a post."""
+        embed_images = item["embed"]["images"]
+        if not embed_images or len(embed_images) == 0:
+            raise NazurinError("No image found")
+        imgs = []
+        for index, pic in enumerate(embed_images):
+            url = pic["fullsize"]
+            thumbnail = pic["thumb"]
+            destination, filename = Bluesky.get_storage_dest(item, pic, index)
+            size = item["record"]["embed"]["images"][index]["image"]["size"]
```
Comment (y-young) [inline nazurin/sites/bluesky/api.py:74]: Seems the size, width and height are not accurate, they're not corresponding to either thumbnail or full-size image, maybe we should just ignore it.

File: docs/site/bluesky.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+:material-lightbulb-on: Optional, defaults to `{rkey}_{index} - {user[display_name]}({user[handle]})`
+
+File name for downloaded images.
+
+### Available Variables
+
+_Only common used ones are listed._
+
+```json
+TODO after code review
```
Comment (y-young) [inline docs/site/bluesky.md:26]: The `context` looks good, please update the document here.

**Review by y-young — CHANGES_REQUESTED** [review]

File: nazurin/sites/bluesky/api.py (hunk 1 above)
Comment (Misaka13514) [reply nazurin/sites/bluesky/api.py:21]: This header is in the API docs but can be removed, I will remove it.

File: nazurin/sites/bluesky/interface.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from time import time
+
+from nazurin.database import Database
+from nazurin.models import Illust
+
+from .api import Bluesky
+from .config import COLLECTION
+
+patterns = [
+    # https://bsky.app/profile/shiratamacaron.bsky.social/post/3kkt7oj5rmw2j
```
Comment (y-young) [inline nazurin/sites/bluesky/interface.py:10]:
```suggestion
    # https://atproto.com/specs/record-key#record-key-syntax
    # https://bsky.app/profile/shiratamacaron.bsky.social/post/3kkt7oj5rmw2j
```

File: nazurin/sites/bluesky/interface.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    r"bsky\.app/profile/([\w\.-]+)/post/([\w\.-~]+)"
```
Comment (y-young) [inline nazurin/sites/bluesky/interface.py:11]:
```suggestion
    r"bsky\.app/profile/([\w\.-]+)/post/([\w\.\-~]+)"
```

**Review by y-young — CHANGES_REQUESTED** [review]
Added some suggestions, seems you didn't allow me to update your PR.

**Comment by Misaka13514** [comment]
(I don't see a checkbox to allow the edit from you, maybe it's because this PR is from an organization?) Thank you for the suggestions, I have applied them.

**Review by y-young — APPROVED** [review]

**Pull request merged**